modules/wallet/database.go

```diff
@@ -82,25 +82,45 @@ func (w *Wallet) threadedDBUpdate() {
 			return
 		}
 		w.mu.Lock()
-		w.syncDB()
+		err := w.syncDB()
 		w.mu.Unlock()
+		if err != nil {
+			// If the database is having problems, we need to close it to
+			// protect it. This will likely cause a panic somewhere when another
+			// caller tries to access dbTx but it is nil.
+			w.log.Severe("ERROR: syncDB encountered an error. Closing the wallet db to protect the database, wallet may crash.", err)
```

Comment:
this message seems redundant, since `syncDB` prints a similar message. I'd keep either one but not both.